### PR TITLE
[Performance] Added the Content-Security-Policy rules

### LINKED ISSUE #3695: [Performance] Fix the `csp-xss` rule reported by lighthouse ci
### Description

Currently the lighthouse ci tests are failing for [csp-xss](https://github.com/layer5io/layer5/actions/runs/4088178421/jobs/7049562479#step:6:40) test 

### Expected Behavior

This assertion test needs to be fixed and a strict content-security-policy needs to be enabled.

Some resources to help solve this issue: 
https://www.isaacsmith.us/blog/2022/add-csp-to-github-pages
https://web.dev/strict-csp/

### Environment:
- Host OS: 
- Browser: 

---
<h2>Contributor Resources and <a href="https://layer5.io/community/handbook">Handbook</a></h2>

The layer5.io website uses Gatsby, React, and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- 📚 See [contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md).
- 🎨 Wireframes and designs for Layer5 site in [Figma](https://www.figma.com/file/5ZwEkSJwUPitURD59YHMEN/Layer5-Designs). 
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io).

Join the Layer5 Community by submitting your [community member form](https://layer5.io/newcomer).

### PR DESCRIPTION
**Pull request by RishabhKodes**
**Description**
Added the csp rules using the `gatsby-plugin-csp`. 

This PR fixes #3695 

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by RishabhKodes** [comment]
CSP Status present now:
image


CSP Status after the fix:
image

File: gatsby-config.js [hunk 1]
```diff
@@ -23,6 +23,20 @@ module.exports = {
...
+      resolve: "gatsby-plugin-csp",
+      options: {
+        disableOnDev: true,
+        reportOnly: false,
+        mergeScriptHashes: false,
+        mergeStyleHashes: false,
+        mergeDefaultDirectives: true,
+        directives: {
+          "default-src": "'self'",  // a fallback for the other CSP directives
+          "script-src": "'self' www.meshery.layer5.io 'unsafe-inline'"
```
Comment (Nikhil-Ladha) [inline gatsby-config.js:36]: You will have to add these as well
```
  "script-src 'self' https://www.google-analytics.com",
  "font-src 'self' https://fonts.googleapis.com https://fonts.gstatic.com",
  "style-src 'self' 'unsafe-inline' https://fonts.googleapis.com",
  "img-src 'self' https://www.google-analytics.com"
```

**Comment by l5io** [comment]
🚀 Preview for commit c768b069307dbbc827248fe8b1342071c1da934c at: https://63f0c0271ee515021dfefe2a--layer5.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit ea64adeaf5f661d4615dce3045e9fa560c8d6360 at: https://63f0ee5239e955222b3f5957--layer5.netlify.app

**Comment by RishabhKodes** [comment]
@Nikhil-Ladha let me know if any other changes are required. Surprisingly I can see the csp-xss security on the lighthouse tab in the preview [link](https://63f0ee5239e955222b3f5957--layer5.netlify.app/), but somehow the lighthouseci is not picking it up and failing.

**Comment by Nikhil-Ladha** [comment]
Yeah, it will fail because the rules that we are setting will not pass the test, but just offer some level of protection.
Though, in the preview, I could see the local fonts still fail to load.
I think let's ignore this test and let it be disabled, we don't wanna invest much effort into this and better fix things that could _actually_ be fixed.

Thanks for your time and effort on this though, @RishabhKodes 🥇

**Pull request closed without merging**